### ISSUE TITLE
ManyToManyField not typed properly when using import alias for related model

### ISSUE THREAD

**Issue by tkoft** [typeddjango/django-stubs#2128]
# Bug report

## What's wrong

Looks like the transformer for ManyToManyField doesn't handle the case for when the related model passed in is aliased. The `to` model is still typed properly, since it's inferred and returned as the FunctionContext `default_return_type`, but any generated `through` model is never typed.

Example:
```
from django.contrib.auth import models as auth_models
from django.db import models

class MyModel(models.Model):
    allowed_groups = models.ManyToManyField(auth_models.Group)
```

I believe [`get_model_from_expression()`](https://github.com/typeddjango/django-stubs/blob/master/mypy_django_plugin/transformers/manytomany.py#L122) gets passed a MemberExpr as such:
```
MemberExpr:11(
  NameExpr(auth_models [django.contrib.auth.models])
  Group [django.contrib.auth.models.Group])
```

But this case is only handled explicitly for the django.contrib.auth.models.User model. In other cases, it returns None, and the transform for the whole field just returns the default value.

## How is that should be

Since normal RelatedFields resolve their related models fine with aliased imports, it seems the manytomany transformer should be able to use the same logic (a la `django_context.get_field_related_model_cls(current_field)`) for resolving both its `to` and `through` models. Though I'm sure there's some complexities around how Django models m2m relations that would require some changes from how `get_field_related_model_cls` works.

## System information

- OS:
- `python` version: 3.8
- `django` version: 4.2.9
- `mypy` version: 1.10
- `django-stubs` version: 5.0
- `django-stubs-ext` version:

**Comment by tkoft** [comment]
Actually, I think the easiest fix is to change this condition to check for RefExpr instead of NameExpr:
https://github.com/typeddjango/django-stubs/blob/4a5b0656be5ccfb961e4e32f4a99756eaf57bb1d/mypy_django_plugin/transformers/manytomany.py#L132

**Comment by flaeppe** [comment]
Thanks for the clear report and a fix, I've opened #2153 with your suggestion
